XMonad/Layout/ConditionalLayout.hs

```diff
@@ -0,0 +1,191 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StrictData #-}
+
+{- |
+   Module       : XMonad.Layout.ConditionalLayout
+   Description  : Conditionally apply layout and layout modifiers.
+   Copyright    : (c) Ivan Malison   <IvanMalison@gmail.com>,
+                      Tomas Janousek <tomi@nomi.cz>
+   License      : BSD
+   Maintainer   : Tony Zorman <soliditsallgood@mailbox.org>
+
+   This module provides conditional variants of 'ModifiedLayout' and 'Choose',
+   so that modifications (specific layouts) are only applied when a particular
+   condition is met.
+-}
+module XMonad.Layout.ConditionalLayout (
+  -- * Usage
+  -- $usage
+
+  -- * Combinators
+  condChoose,
+  conditional,
+
+  -- * Building Conditions
+  ModifierCondition (shouldApply),
+) where
+
+import XMonad hiding (hide)
+import XMonad.Layout.LayoutModifier
+import XMonad.Prelude
+import XMonad.StackSet (Workspace)
+import qualified XMonad.StackSet as W
+
+{- $usage
+
+You can use this module by importing it in your @xmonad.hs@
+
+> import XMonad.Layout.ConditionalLayout
+
+and writing an appropriate condition. Then, when 'conditional' is used,
+a layout modifier will be applied whenever the condition is true.
+Alternatively, 'condChoose' can decide which of two given layouts is to
+be applied.
+
+Defining a condition works by creating a new type, and making it an
+instance of 'ModifierCondition'. For example, the following condition
+checks whether there are a maximum of two windows on the current
+workspace:
+
+> import Data.List
+>
+> import XMonad
+> import XMonad.Layout.ConditionalLayout
+> import qualified XMonad.StackSet as W
+>
+> data IfMax2 = IfMax2 deriving (Read, Show)
+>
+> instance ModifierCondition IfMax2 where
+>   shouldApply _ wsId = do
+>     wins <- gets $ find ((wsId ==) . W.tag) . W.workspaces . windowset
+>     let ws = W.integrate' . W.stack <$> wins
+>     pure $ Just 2 >= (length <$> ws)
+
+the @IfMax2@ type can now be used with the provided combinators. To
+apply tabs, one would write
+
+> import XMonad.Layout.Tabbed
+>
+> main :: IO ()
+> main = xmonad $ def
+>   { terminal   = "urxvt"
+>   , layoutHook = conditional IfMax2 (addTabsAlways shrinkText def) $ layoutHook def
+>   }
+
+alternatively, to conditionally switch between two layouts:
+
+>                  -- Full for workspaces with more than 4 windows, Tall otherwise.
```

Comment:
Isn't this "more than two"? Or is the condition applied twice somehow?

Reply:
Ah, good catch, this must still be a remnant from an earlier version